xgcm/grid.py

```diff
@@ -1352,11 +1367,11 @@ def get_metric(self, array, axes):
                 if metric_dims.issubset(array_dims):
                     metric_vars = mv
                     break
-            # if metric_vars is None:
-            #     # Condition 4: interpolate metric with matching axis to desired dimensions
-            #     metric_vars = self.interp_like(mv, array)
+            if metric_vars is None:
+                # Condition 2: interpolate metric with matching axis to desired dimensions
+                metric_vars = self.interp_like(mv, array, boundary, fill_value)
```

Comment:
So what happens when the user does not set boundary or fill_value? Shoud we set 'reasonable defaults'?

Reply:
Line 1320: `boundary=None, fill_value=None`

Reply:
> boundary or fill_value? Shoud we set 'reasonable defaults'?

I prefer making the user specify defaults in the Grid constructor. We already do this with boundary.

I think xgcm should encourage the pattern of "setup your grid object for all possible cases, then do analysis".

Reply:
So because boundary and fill_value are already specified with Grid:
```python
class Grid:
    """
    An object with multiple :class:`xgcm.Axis` objects representing different
    independent axes.
    """

    def __init__(
        self,
        ds,
        check_dims=True,
        periodic=True,
        default_shifts={},
        face_connections=None,
        coords=None,
        metrics=None,
        boundary=None,
        fill_value=None,
    ):
```

Those values will pass on to `get_metric` and I can just use:
```python
def get_metric(self, array, axes, boundary, fill_value):
```
instead of:
```python
def get_metric(self, array, axes, boundary=None, fill_value=None):
```

Reply:
No you want
``` python
def get_metric(self, array, axes, boundary=None, fill_value=None):	
```
The first version (with no default values) requires the user to specify `boundary` and `fill_value` always even if default `boundary` and `fill_value` is set on the `Axis` object.

We want the second version because that allows the user to override at the method-level if necessary but when `boundary is None` (the default) `_get_neighbour_binary_func` (or whatever) will use `ax.boundary`
https://github.com/xgcm/xgcm/blob/f61edc455cc4a1c5d35e7f9ffaf88f408f333f65/xgcm/grid.py#L281-L285

Reply:
Alright!

> I prefer making the user specify defaults in the Grid constructor. We already do this with boundary.
> I think xgcm should encourage the pattern of "setup your grid object for all possible cases, then do analysis".

I do wonder what you meant by the statement above in relation to get_metric.

Reply:
I suggest to just hardcode `self.interp_like(mv, array, 'extend', None)` here for simplicity. What do you think about that? Would eliminate all the fiddling with boundary and fill_val in the other methods.